src/Test/TestQueries.cs

```diff
@@ -0,0 +1,430 @@
+                        }
+                        output += ")";
+                    }
+                    output += " FROM " + tableName;
+                    if (isSumaarizeByClausePresent)
+                    {
+                        output += " GROUP BY ";
+                        for(int i = 0; i < summarizeByColumns.Count; i++)
+                        {
+                            output += summarizeByColumns[i];
+                            if (i != summarizeByColumns.Count - 1) output += ",";
+                        }
+                    }
+                    return output;
+                }
+                if (isSumaarizeOperatorPresent & isSumaarizeByClausePresent & !isFunctionCallPresent)
+                {
+                    output += "SELECT DISTINCT ";
+                    for (int i = 1; i < allTokenNames.Count; i++)
+                    {
+                        output += allTokenNames[i];
+                        if (i != allTokenNames.Count - 1) output += ",";
+                    }
+                    output += " FROM " + tableName;
+                    return output;
+                }
+                /*if (isSumaarizeOperatorPresent & isSumaarizeByClausePresent & isFunctionCallPresent)
+                {
+                    output += "SELECT ";
+                    if (onlyFunctionNames.Contains(" avg"))
+                    {
+                        int index = onlyFunctionNames.IndexOf(" avg");
+                        for (int i = 0; i < summarizeByColumns.Count; i++)
+                        {
+                            if (i == summarizeByColumns.Count - 1)
+                            {
+                                output += summarizeByColumns[i];
+                            }
+                            else output += summarizeByColumns[i] + ",";
+                        }
+                        var list = allFunctioCallNames[index];
+                        output += " ," +  SQL.average + "(";
+                        for (int i = 1; i < list.Count; i++)
+                        {
+                            if (i == list.Count - 1)
+                            {
+                                output += list[i];
+                            }
+                            else output += list[i] + ",";
+                        }
+                        output += ") FROM " + tableName;
+                        output += " GROUP BY ";
+                        for (int i = 0; i < summarizeByColumns.Count; i++)
+                        {
+                            if (i == summarizeByColumns.Count - 1)
+                            {
+                                output += summarizeByColumns[i];
+                            }
+                            else output += summarizeByColumns[i] + ",";
+                        }
+                    }
+                    return output;
+                }*/
+            }
+            if (numPipeExpression == 2)
```

Comment:
Here, we can make it more generic. Like, instead we should start thinking about to build SQL query from n expressions given. This hardcoding wont scale for complex queries. This is good for now, but next step should be in that direction